tests/unit/store/push-test.js

```diff
@@ -606,6 +606,69 @@ testInDebug(
   }
 );
 
+testInDebug(
+  'Calling push with a link for a non async relationship should not reset an existing relationship',
+  function(assert) {
+    Person.reopen({
```

Comment:
could you write a test that

- does not use `reopen`
- does not use `run`

To do this you may need to make the test `async`.  In some cases you may also want to `await settled()` to ensure relationship state is fully settled post-push, although in this case it should not matter.